src/JsonApiSerializer/JsonConverters/ResourceIdentifierConverter.cs

```diff
@@ -298,15 +298,23 @@ private object ReadJsonAsResourceObject(ForkableJsonReader reader, Type objectTy
                 var contract = (JsonObjectContract)jsonApiContractResolver.ResolveContract(objectType);
 
                 resourceObject = ReaderUtil.CreateObject(serializationData, objectType, reference.Type, serializer);
+                var propertyAttr = objectType.GetTypeInfo().GetCustomAttribute<JsonApiProperties>() ?? new JsonApiProperties();
 
                 // for placeholders we will just read the top level properties
                 // it is unlikely to have attributes/relationships present
                 foreach (var propName in ReaderUtil.IterateProperties(reader))
                 {
+                    // If an "Id" override attribute is present then deserialize to the correct object property
+                    var pName = propName;
+                    if (propName == PropertyNames.Id && propertyAttr.Id != string.Empty)
```

Comment:
```suggestion
                    if (propName == PropertyNames.Id && !string.IsNullOrEmpty(propertyAttr.Id))
```

Reply:
The reason I'm not using null check explicitly here is because I wanted to treat `null` values as forcing the serializer to treat the `Id` and `Type` fields as a regular field. As such, with the attribute either it will be a non-empty string to new column, or `null` to designate that the class has no dedicated `Id`/`Type` property.

I'm open to suggestion so as how to accomplish it differently - I was trying to avoid adding additional Attribute properties to set.